test/e2e/Makefile

```diff
@@ -0,0 +1,138 @@
+	-./kind-install-for-capd.sh
+
+## --------------------------------------
+## Manifests
+## --------------------------------------
+
+.PHONY: manifests
+manifests: 
+	d=$$(date +%s) \
+	&& IMG=$(IMG) make -C $(REPO_ROOT) manifests \
+	&& echo "This step took $$(($$(date +%s)-d)) seconds"
+
+## --------------------------------------
+## Templates
+## --------------------------------------
+
+CLOUDSTACK_TEMPLATES := $(REPO_ROOT)/test/e2e/data/infrastructure-cloudstack
+
+.PHONY: cluster-templates
+cluster-templates: cluster-templates-v1beta1
+
+.PHONY: cluster-templates-v1beta1
+cluster-templates-v1beta1: $(KUSTOMIZE)
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-kcp-remediation > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-kcp-remediation.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-md-remediation > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-md-remediation.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-zone > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-zone.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-account > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-account.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-domain > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-domain.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-template > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-template.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-cp-offering > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-cp-offering.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-worker-offering > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-worker-offering.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-insufficient-compute-resources > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-insufficient-compute-resources.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-node-drain > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-node-drain.yaml
+
+## --------------------------------------
+## Docker build and push to local registry
+## --------------------------------------
+
+.PHONY: docker-build-and-push
+docker-build-and-push:
```

Comment:
This is useful enough it would be nice to place it in the main makefile and call it something like `local-build-and-push`.